src/composables/useFormatDate.ts

```diff
@@ -0,0 +1,126 @@
+// libraries
+import { date } from 'quasar';
+
+// composables
+import { i18n } from 'src/boot/i18n';
+
+// enums
+enum TimeState {
+  INVALID = 'INVALID',
+  NOW = 'NOW',
+  TODAY = 'TODAY',
+  YESTERDAY = 'YESTERDAY',
+  PAST7DAYS = 'PAST7DAYS',
+  PAST = 'PAST',
+}
+
+export const useFormatDate = () => {
+  /**
+   * Returns a label for a given date relative to the current date.
+   *
+   * It only accepts past dates. Returns empty string for future dates.
+   * If given date is the same as today, the label will show
+   * "X hours ago".
+   * If given date is within 7 days of the current date, the label will show
+   * "Yesterday", or the name of the day of the week and time.
+   * Otherwise, the label will show the date in the format "D. MMM". and time.
+   *
+   * Used for forum post list.
+   *
+   * @param dateString - The date string to be formatted
+   * @return The formatted date name
+   */
+  const formatDateTimeLabel = (dateString: string): string => {
+    if (!date.isValid(dateString)) return '';
+    const nowStamp = new Date();
+    const timeStamp = new Date(dateString);
+    const state: TimeState = getTimeState({ nowStamp, timeStamp });
+
+    switch (state) {
+      case TimeState.NOW:
+        return i18n.global.t('time.lessThanAnHourAgo');
+      case TimeState.TODAY:
+        return labelToday({ nowStamp, timeStamp });
+      case TimeState.YESTERDAY:
+        return labelYesterday({ timeStamp });
+      case TimeState.PAST7DAYS:
+        return labelPast7Days({ timeStamp });
+      case TimeState.PAST:
+        return date.formatDate(timeStamp, 'D. MMM., HH:mm');
+      default:
+        return '';
+    }
+  };
+
+  /**
+   * Returns the "state" based on date comparison.
+   */
+  function getTimeState({
+    nowStamp,
+    timeStamp,
+  }: {
+    nowStamp: Date;
+    timeStamp: Date;
+  }): TimeState {
+    let state = TimeState.PAST;
```

Comment:
Better readibility/simplicity with `switch()` statement,  but tests failed...

```
    let state;
    switch (true) {
      case date.getDateDiff(nowStamp, timeStamp, 'days') < 7:
        state = TimeState.PAST7DAYS;
        break;
      case date.getDateDiff(nowStamp, timeStamp, 'days') === 1:
        state = TimeState.YESTERDAY;
        break;
      case date.isSameDate(nowStamp, timeStamp, 'day'):
        state = TimeState.TODAY;
        break;      
      case date.getDateDiff(nowStamp, timeStamp, 'hours') < 1:
        state = TimeState.NOW;
        break;
      default:
          state = TimeState.PAST;
    }
    return state
```

Reply:
> Better readibility/simplicity with `switch()` statement, but tests failed... It is seems these conditions are much more complicated. Why?

In that case the order of conditions needs to be reversed since `break` works as an early return statement.
Updated.